src/serial/SerialBar.tsx

```diff
@@ -33,12 +34,25 @@ const SerialBar = ({
   background,
   ...props
 }: SerialBarProps) => {
+  const logging = useLogging();
   const handleExpandCollapseClick = useCallback(() => {
+    logging.event({
+      type: compact ? "serial-expand" : "serial-collapse",
+    });
     onSizeChange(compact ? "open" : "compact");
-  }, [compact, onSizeChange]);
+  }, [compact, onSizeChange, logging]);
   const intl = useIntl();
   const helpDisclosure = useDisclosure();
   const traceback = useDeviceTraceback();
+  const [prevTraceback, setPrevTraceback] = useState<Traceback | undefined>();
+  useEffect(() => {
+    if (traceback && !prevTraceback) {
+      logging.event({
+        type: "serial-traceback",
+      });
+    }
+    setPrevTraceback(traceback);
+  }, [logging, traceback, prevTraceback, setPrevTraceback]);
```

Comment:
This feels clumsy. Is there a better way to get the previous traceback? We could change the hook, but it seems like more effort than it's worth when we only need the previous traceback here.

Reply:
Improved after discussion with @microbit-matt-hillsdon. See [`fa77951`](https://github.com/microbit-foundation/python-editor-next/pull/517/commits/fa779517cc9338cdec9e6094cc9d18589d94e0b4).